mapiproxy/libmapistore/backends/python/rest.py

```diff
@@ -519,10 +568,14 @@ def get_properties(self, properties):
         properties['PidTagDisplayName'] = str(self.properties['PidTagDisplayName'])
         properties['PidTagComment'] = str(self.properties['PidTagComment'])
         properties['PidTagFolderType'] = mapistore.FOLDER_GENERIC
-        properties['PidTagAccess'] = PidTagAccessFlag.Read|PidTagAccessFlag.HierarchyTable|PidTagAccessFlag.ContentsTable|PidTagAccessFlag.AssocContentsTable
+        properties['PidTagAccess'] = (PidTagAccessFlag.Modify |
+                                      PidTagAccessFlag.Read |
+                                      PidTagAccessFlag.HierarchyTable |
+                                      PidTagAccessFlag.ContentsTable |
+                                      PidTagAccessFlag.AssocContentsTable)
         properties['PidTagCreationTime'] = float((datetime.now(tz=timezone('Europe/Madrid')) - timedelta(hours=1)).strftime("%s.%f"))
         properties['PidTagLastModificationTime'] = properties['PidTagCreationTime']
-        properties["PidTagChangeKey"] = bytearray(uuid.uuid1().bytes + '\x00\x00\x00\x00\x00\x01')
+        properties["PidTagChangeKey"] = bytearray(uuid.UUID('082152cf-ae39-44a1-b11f-a1e1af3bac08').bytes + uuid.uuid1().bytes[0:6])
```

Comment:
This uuid is specific to your deployment. I also agree that the initial approach is not better, however, randomness for randomness I would like to keep the initial approach.

However add a /* FIXME: Generate PidTagChangeKey based on PidTagChangeNumber */